test/math/ring_signature.cpp

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2011-2017 libbitcoin developers (see AUTHORS)
+ *
+ * This file is part of libbitcoin.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <random>
+#include <boost/test/unit_test.hpp>
+#include <bitcoin/bitcoin.hpp>
+
+using namespace bc;
+
+data_chunk new_seed()
+{
+    data_chunk seed(10);
+    pseudo_random_fill(seed);
+    return seed;
+}
+
+ec_secret new_key()
+{
+    const data_chunk seed = new_seed();
+    const wallet::hd_private key(seed);
+    return key.secret();
+}
+
+template <typename SecretsIterator>
+point_list extract_public_keys(SecretsIterator start, SecretsIterator end)
+{
+    point_list ring;
+    for (auto it = start; it != end; ++it)
+    {
+        ec_compressed public_key;
+        bool rc = secret_to_public(public_key, *it);
+        BITCOIN_ASSERT(rc);
```

Comment:
Remove result suppression (factored out of NDEBUG builds)) and rely on test helpers.
```cpp
BOOST_REQUIRE(secret_to_public(public_key, *it));
```